data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/BinlogEventListener.java

```diff
@@ -223,6 +233,8 @@ private void handleRowChangeEvent(com.github.shyiko.mysql.binlog.event.Event eve
         final List<String> primaryKeys = tableMetadata.getPrimaryKeys();
         final long eventTimestampMillis = event.getHeader().getTimestamp();
 
+        final BufferAccumulator<Record<Event>> bufferAccumulator = BufferAccumulator.create(buffer, DEFAULT_BUFFER_BATCH_SIZE, BUFFER_TIMEOUT);
```

Comment:
Why is this added instead of creating on constructor ?

Reply:
As @dlvenable mentioned below, it's not thread safe.